File: WalletTracker/app/src/main/java/com/chtima/wallettracker/models/Category.java
package com.chtima.wallettracker.models;


import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Objects;


@Entity(tableName = "categories")
public class Category {

    @PrimaryKey(autoGenerate = true)
    public long id;

    public String title;
    public String icon;
    public CategoryType categoryType;

    public Category(String title, String icon, CategoryType categoryType) {
        this.title = title;
        this.icon = icon;
        this.categoryType = categoryType;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Category category = (Category) o;
        return id == category.id && Objects.equals(title, category.title) && Objects.equals(icon, category.icon) && categoryType == category.categoryType;
    }

    public enum CategoryType{
        INCOME,
        EXPENSE,
        MIX
    }
}

File: WalletTracker/app/src/main/java/com/chtima/wallettracker/models/Category.kt
package com.chtima.wallettracker.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Objects


@Entity(tableName = "categories")
class Category(var title: String, var icon: String, var categoryType: CategoryType) {

    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val category = o as Category
        return id == category.id && Objects.equals(title, category.title) && Objects.equals(
            icon,
            category.icon
        ) && categoryType == category.categoryType
    }

    override fun hashCode(): Int {
        var result = title.hashCode()
        result = 31 * result + icon.hashCode()
        result = 31 * result + categoryType.hashCode()
        result = 31 * result + id.hashCode()
        return result
    }

    enum class CategoryType {
        INCOME,
        EXPENSE,
        MIX
    }

}